RecoTracker/MeasurementDet/plugins/MaskedMeasurementTrackerEventProducer.cc

```diff
@@ -14,21 +14,35 @@ class dso_hidden MaskedMeasurementTrackerEventProducer final : public edm::strea
 
       typedef edm::ContainerMask<edmNew::DetSetVector<SiStripCluster> > StripMask;
       typedef edm::ContainerMask<edmNew::DetSetVector<SiPixelCluster> > PixelMask;
+      typedef edm::ContainerMask<edmNew::DetSetVector<Phase2TrackerCluster1D> > Phase2OTMask;
 
       edm::EDGetTokenT<MeasurementTrackerEvent> src_;
 
       edm::EDGetTokenT<StripMask> maskStrips_;
       edm::EDGetTokenT<PixelMask> maskPixels_;
+      edm::EDGetTokenT<Phase2OTMask> maskPhase2OTs_;
+
+      bool skipClusters_;
+      bool phase2skipClusters_;
 };
 
 
 MaskedMeasurementTrackerEventProducer::MaskedMeasurementTrackerEventProducer(const edm::ParameterSet &iConfig) :
     src_(consumes<MeasurementTrackerEvent>(iConfig.getParameter<edm::InputTag>("src")))
 {
-    edm::InputTag clustersToSkip = iConfig.getParameter<edm::InputTag>("clustersToSkip");
-    maskStrips_ = consumes<StripMask>(clustersToSkip);
-    maskPixels_ = consumes<PixelMask>(clustersToSkip);
-
+    //FIXME:temporary solution in order to use this class for both phase0/1 and phase2
+    if (iConfig.existsAs<edm::InputTag>("clustersToSkip")) {
```

Comment:
it's the first instance of existsAs in this producer.
Define fillDescriptions instead